infor-design/enterprise-wc
# 930 - Improve IdsToolbarMoreActions menu

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR makes a few improvements to the IdsToolbarMoreActions component (primarily in the IdsPopupMenu part):
- Nested submenus inside the More Actions menu no longer close when activated
- Keyboard functionality to IdsPopupMenu overall has been restored
- Fixed hover/mouseover highlighting behavior to work correctly when mixed with keyboard control of the menu (you can now interchangeably navigate menus by dragging the mouse to a menu item, then use the keyboard to navigate away like what was possible in 4.x)
- Fixed premature menu closing in some cases when moving the mouse from a top-level menu to a submenu
- Made menu icon spacing/alignment depend on whether or not any menu icons are visible (updates when toolbar overflow changes)

**Related github/jira issue (required)**:
Closes #930

**Steps necessary to review your pull request (required)**:
Pull/Build/Run, then test the following cases:

**Test submenu accessibility:**
- Open http://localhost:4300/ids-toolbar/overflowed.html
- Shrink the browser window so that only "Icon Button 4" and "Text Button 5" are displayed, and other buttons are pushed to overflow
- Open the "..." menu > "Menu Button 2" > "More Items".  The menu should be accessible.

**Test keyboard functionality:**
- Open http://localhost:4300/ids-toolbar/overflowed.html and shrink the window similar to above
- Use the tab key to navigate to the IdsToolbar, then use the arrow keys to navigate to the "...".  Press enter, and use the arrow keys to navigate the more actions menu.  Use the right arrow to open submenus, and the left arrow to close submenus.  Navigation with keyboard should make sense

**Test highlighting:**
- Open http://localhost:4300/ids-toolbar/overflowed.html and shrink the window similar to above
- Use the keyboard to open the menu, navigate two menu items down, and press "escape" to close
- Re-open the menu with Enter or Spacebar.  The item highlighted should be the same as when you closed

**Test keyboard/mouse nav mixing**
- Open http://localhost:4300/ids-toolbar/overflowed.html and shrink the window similar to above
- Use the mouse to open the more actions menu, and pick any menu item to hover
- Without closing the menu, switch to keyboard navigation.  The next/previous highlighted items should be relative to the mouse-hovered menu item.

Also, smoke-test the following examples to make sure they are all still working:
- http://localhost:4300/ids-popup-menu/example.html
- http://localhost:4300/ids-popup-menu/selected-state.html

**Included in this Pull Request**:
